src/snapred/backend/recipe/algorithm/MantidSnapper.py

```diff
@@ -254,3 +256,18 @@ def cleanup(self):
             self._prog_reporter.report(self._endrange, "Done")
         self._progressCounter = 0
         self._algorithmQueue = []
+
+    def _addWorkspaceToDict(self, prop, val, name):
+        if prop == "OutputWorkspace":
+            if self.mtd.doesExist(val):
+                self.workspaceDict[val] = (self.mtd[val], name)
```

Comment:
You are asking for trouble here. The workspace references have a habit of getting invalidated because ADS (called `mtd` here) doesn't know to update viewers of workspaces. There is a framework for that, but it would take some time to find it.